pkg/workflows/wasm/host/module.go

```diff
@@ -110,7 +110,17 @@ type ModuleConfig struct {
 	Determinism *DeterminismConfig
 }
 
-type Module struct {
+type Module interface {
```

Comment:
@bolekk Did you consider having a ModuleV1 and ModuleV2 rather than an interface that combines the two? They aren't really interchangeable so it would make sense to rely on the type system to enforce using the right one in the right context.

You only need to know about the difference in one place (the workflow registry syncer); after that you should be guaranteed to be dealing with one or the other depending on the context.

Reply:
Makes sense, let me see if this fits well with the draft that Ryan has for V2 implementation (keeping the same underlying object).

Reply:
Done.